src/System.Numerics.Matrices/src/System/Numerics/Matrix4x4.cs

```diff
@@ -0,0 +1,435 @@
+        /// <param name="m24">The column 2, row 4 value</param>
+        /// <param name="m34">The column 3, row 4 value</param>
+        /// <param name="m44">The column 4, row 4 value</param>
+        public Matrix4x4(double m11, double m21, double m31, double m41, 
+                         double m12, double m22, double m32, double m42, 
+                         double m13, double m23, double m33, double m43, 
+                         double m14, double m24, double m34, double m44)
+        {
+            M11 = m11; M21 = m21; M31 = m31; M41 = m41; 
+            M12 = m12; M22 = m22; M32 = m32; M42 = m42; 
+            M13 = m13; M23 = m23; M33 = m33; M43 = m43; 
+            M14 = m14; M24 = m24; M34 = m34; M44 = m44; 
+        }
+
+        /// <summary>
+        /// Initialized a Matrix4x4 with all values set to the same value
+        /// </summary>
+        /// <param name="value">The value to set all values to</param>
+        public Matrix4x4(double value)
+        {
+            M11 = M21 = M31 = M41 = 
+            M12 = M22 = M32 = M42 = 
+            M13 = M23 = M33 = M43 = 
+            M14 = M24 = M34 = M44 = value;
+        }
+
+        public double M11;
+        public double M21;
+        public double M31;
+        public double M41;
+        public double M12;
+        public double M22;
+        public double M32;
+        public double M42;
+        public double M13;
+        public double M23;
+        public double M33;
+        public double M43;
+        public double M14;
+        public double M24;
+        public double M34;
+        public double M44;
+
+        public unsafe double this[int col, int row]
+        {
+            get
+            {
+                if (col < 0 || col >= ColumnCount)
+                    throw new ArgumentOutOfRangeException(nameof(col), $"Expected greater than or equal to 0 and less than {ColumnCount}, found {col}.");
+                if (row < 0 || row >= RowCount)
+                    throw new ArgumentOutOfRangeException(nameof(row), $"Expected greater than or equal to 0 and less than {RowCount}, found {row}.");
+
+                fixed (Matrix4x4* p = &this)
+                {
+                    double* d = (double*)p;
+                    return d[row * ColumnCount + col];
+                }
+            }
+            set
+            {
+                if (col < 0 || col >= ColumnCount)
+                    throw new ArgumentOutOfRangeException(nameof(col), $"Expected greater than or equal to 0 and less than {ColumnCount}, found {col}.");
+                if (row < 0 || row >= RowCount)
+                    throw new ArgumentOutOfRangeException(nameof(row), $"Expected greater than or equal to 0 and less than {RowCount}, found {row}.");
+
+                fixed (Matrix4x4* p = &this)
+                {
+                    double* d = (double*)p;
+                    d[row * ColumnCount + col] = value;
+                }
+            }
```

Comment:
Anyone know if this alignment, which works well on x86, will work on ARM and other platforms?